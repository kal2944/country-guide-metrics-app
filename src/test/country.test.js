import { render } from '@testing-library/react';
import * as reactRedux from 'react-redux';
import Home from '../mock/home';

jest.mock('react-redux');

const mockDispatch = jest.fn();

beforeAll(() => {
  reactRedux.useDispatch = jest.fn().mockImplementation(() => mockDispatch);
});

describe('Test for Home Page', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render the Home Page', async () => {
    const countries = [
      {
        name: 'Afghanistan',
        topLevelDomain: ['.af'],
        alpha2Code: 'AF',
        alpha3Code: 'AFG',
        callingCodes: ['93'],
        capital: 'Kabul',
        altSpellings: ['AF', 'Afġānistān'],
        subregion: 'Southern Asia',
        region: 'Asia',
        population: 40218234,
        latlng: [33.0, 65.0],
        demonym: 'Afghan',
        area: 652230.0,
        timezones: ['UTC+04:30'],
        borders: ['IRN', 'PAK', 'TKM', 'UZB', 'TJK', 'CHN'],
        nativeName: 'افغانستان',
        numericCode: '004',
        flags: {
          svg: 'https://upload.wikimedia.org/wikipedia/commons/5/5c/Flag_of_the_Taliban.svg',
          png: 'https://upload.wikimedia.org/wikipedia/commons/thumb/5/5c/Flag_of_the_Taliban.svg/320px-Flag_of_the_Taliban.svg.png',
        },
        currencies: [
          {
            code: 'AFN',
            name: 'Afghan afghani',
            symbol: '؋',
          },
        ],
        languages: [
          {
            iso639_1: 'ps',
            iso639_2: 'pus',
            name: 'Pashto',
            nativeName: 'پښتو',
          },
          {
            iso639_1: 'uz',
            iso639_2: 'uzb',
            name: 'Uzbek',
            nativeName: 'Oʻzbek',
          },
          {
            iso639_1: 'tk',
            iso639_2: 'tuk',
            name: 'Turkmen',
            nativeName: 'Türkmen',
          },
        ],
        translations: {
          br: 'Afghanistan',
          pt: 'Afeganistão',
          nl: 'Afghanistan',
          hr: 'Afganistan',
          fa: 'افغانستان',
          de: 'Afghanistan',
          es: 'Afganistán',
          fr: 'Afghanistan',
          ja: 'アフガニスタン',
          it: 'Afghanistan',
          hu: 'Afganisztán',
        },
        flag: 'https://upload.wikimedia.org/wikipedia/commons/5/5c/Flag_of_the_Taliban.svg',
        regionalBlocs: [
          {
            acronym: 'SAARC',
            name: 'South Asian Association for Regional Cooperation',
          },
        ],
        cioc: 'AFG',
        independent: true,
      },
      {
        name: 'Åland Islands',
        topLevelDomain: ['.ax'],
        alpha2Code: 'AX',
        alpha3Code: 'ALA',
        callingCodes: ['358'],
        capital: 'Mariehamn',
        altSpellings: ['AX', 'Aaland', 'Aland', 'Ahvenanmaa'],
        subregion: 'Northern Europe',
        region: 'Europe',
        population: 28875,
        latlng: [60.116667, 19.9],
        demonym: 'Ålandish',
        area: 1580.0,
        timezones: ['UTC+02:00'],
        nativeName: 'Åland',
        numericCode: '248',
        flags: {
          svg: 'https://flagcdn.com/ax.svg',
          png: 'https://flagcdn.com/w320/ax.png',
        },
        currencies: [
          {
            code: 'EUR',
            name: 'Euro',
            symbol: '€',
          },
        ],
        languages: [
          {
            iso639_1: 'sv',
            iso639_2: 'swe',
            name: 'Swedish',
            nativeName: 'svenska',
          },
        ],
        translations: {
          br: 'Åland',
          pt: 'Ilhas de Aland',
          nl: 'Ålandeilanden',
          hr: 'Ålandski otoci',
          fa: 'جزایر الند',
          de: 'Åland',
          es: 'Alandia',
          fr: 'Åland',
          ja: 'オーランド諸島',
          it: 'Isole Aland',
          hu: 'Åland-szigetek',
        },
        flag: 'https://flagcdn.com/ax.svg',
        regionalBlocs: [
          {
            acronym: 'EU',
            name: 'European Union',
          },
        ],
        independent: false,
      },
    ];

    const tree = render(<Home countries={countries} />);
    expect(tree).toMatchSnapshot();
  });
});
