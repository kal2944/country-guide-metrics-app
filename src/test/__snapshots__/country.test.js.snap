// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Home Page should render the Home Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="home-header"
        >
          <div
            class="search"
          >
             
            <input
              name="searchItem"
              placeholder="Search for a country"
              type="text"
              value=""
            />
          </div>
        </header>
        <div
          class="countries-grid"
        >
          <button
            class="country"
            type="button"
          >
            <div
              class="image"
            >
              <img
                alt="Country Flag"
                class="country-img"
                src="https://upload.wikimedia.org/wikipedia/commons/5/5c/Flag_of_the_Taliban.svg"
              />
            </div>
            <div
              class="details"
            >
              <h2>
                Afghanistan
              </h2>
              <p>
                Population:
                <span
                  class="span"
                >
                  40218234
                </span>
              </p>
            </div>
          </button>
          <button
            class="country"
            type="button"
          >
            <div
              class="image"
            >
              <img
                alt="Country Flag"
                class="country-img"
                src="https://flagcdn.com/ax.svg"
              />
            </div>
            <div
              class="details"
            >
              <h2>
                Åland Islands
              </h2>
              <p>
                Population:
                <span
                  class="span"
                >
                  28875
                </span>
              </p>
            </div>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="home-header"
      >
        <div
          class="search"
        >
           
          <input
            name="searchItem"
            placeholder="Search for a country"
            type="text"
            value=""
          />
        </div>
      </header>
      <div
        class="countries-grid"
      >
        <button
          class="country"
          type="button"
        >
          <div
            class="image"
          >
            <img
              alt="Country Flag"
              class="country-img"
              src="https://upload.wikimedia.org/wikipedia/commons/5/5c/Flag_of_the_Taliban.svg"
            />
          </div>
          <div
            class="details"
          >
            <h2>
              Afghanistan
            </h2>
            <p>
              Population:
              <span
                class="span"
              >
                40218234
              </span>
            </p>
          </div>
        </button>
        <button
          class="country"
          type="button"
        >
          <div
            class="image"
          >
            <img
              alt="Country Flag"
              class="country-img"
              src="https://flagcdn.com/ax.svg"
            />
          </div>
          <div
            class="details"
          >
            <h2>
              Åland Islands
            </h2>
            <p>
              Population:
              <span
                class="span"
              >
                28875
              </span>
            </p>
          </div>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
